modules/4337/contracts/Safe4337Module.sol

```diff
@@ -212,9 +212,21 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
     }
 
     /**
-     * @dev Checks if the signatures length is correct and does not contain additional bytes. The function does not
+     * @notice Checks if the signatures length is correct and does not contain additional bytes. The function does not
      * check the integrity of the signature encoding, as this is expected to be checked by the {Safe} implementation
      * of {checkSignatures}.
+     * @dev Safe account has two types of signatures: EOA and Smart Contract signatures. While the EOA signature is
+     * fixed in size, the Smart Contract signature can be of arbitrary length. Safe encodes the Smart Contract
+     * signature length in the signature data. If appropriate length checks are not performed during the signature
+     * verification then a malicious bundler can pad additional bytes to the signatures data and make the account pay
+     * more gas than needed for user operation validation and reach the verificationGasLimit.
+     * `_checkSignaturesLength` function checks for the presence of any padded bytes to the `signature` data.
```

Comment:
This is _technically_ not true (and we could get a bug submission with a counter example).

For example:

```
0000000000000000000000001111111111111111111111111111111111111111    # signer 1
00000000000000000000000000000000000000000000000000000000000000c3    # offset 1&2
00                                                                  # contract signature
0000000000000000000000002222222222222222222222222222222222222222    # signer 2
00000000000000000000000000000000000000000000000000000000000000c3    # offset 1&2
00                                                                  # contract signature
0000000000000000000000003333333333333333333333333333333333333333    # signer 3
00000000000000000000000000000000000000000000000000000000000000e8    # offset 3
00                                                                  # contract signature
0000000000000000000000000000000000000000000000000000000000000001 aa # signature 1 & 2
deadbeef                                                            # padding
0000000000000000000000000000000000000000000000000000000000000001 bb # signature 3
```

This will pass the length check with padding.

I would argue that what this does is:
> `_checkSignaturesLength` ensures that the `signatures` data cannot be longer than the canonical encoding of Safe signatures, thus setting a strict upper bound on how long the `signatures` bytes can be, greatly limiting a malicious bundler's ability to pad signature bytes.

Reply:
https://github.com/safe-global/safe-modules/pull/471/commits/86b8c96c64bf700c5f3dcab99e3d391342b3fa39